### DISCUSSION TITLE
How to convert JPG to 10bit AVIF?

### DISCUSSION THREAD

**Discussion by LITUATUI** [ImageMagick/ImageMagick#4932] [Help] [answered]
I usually use Gimp to convert my JPG images to 10bit AVIF, but ImageMagick is much better for batch conversions.

Is it even possible to do it with ImageMagick?

Thanks!

**Comment by snibgo** [comment]
As far as I know, IM can only create 8 bit/channel AVIF, not 10 bit/channel.

**Comment by LITUATUI** [comment]
That's what I thought, it's a shame as 10-bit encoding is more efficient.

**Comment by dlemstra** [comment]
Have you tried using `-depth 10`?

**Reply by LITUATUI** [reply]
> Have you tried using `-depth 10`?

Yes, I did. 

`magick mogrify -format avif -depth 10 *.jpg`

It produces 8-bit AVIF images.

**Comment by urban-warrior** [comment]
We don't write 10-bit AVIF images just yet-- although ImageMagick can support 10-bit.  Its just a matter of taking the time to accommodate 10-bit.  Currently we do not have an ETA on when this capability will be exported.

**Reply by LITUATUI** [reply]
Thanks. I'll read every changelog from now on... :eyes: 

Nowadays AVIF and 10-bit is the best combination for website images by far.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by urban-warrior** [comment, marked as answer]
Look for the ImageMagick 7.1.0-28 release within a few days.  It supports reading and writing 10 & 12-bit AVIF images.

**Reply by LITUATUI** [reply]
That's awesome. Thanks!

**Reply by LITUATUI** [reply]
The update already arrived to Arch distros. I'm using EndeavourOS and it works. I now get 10-bit AVIF files, thanks!

`magick mogrify -format avif -depth 10 *.jpg`

Running the magick identify command:
`AVIF 900x583 900x583+0+0 10-bit sRGB 52156B 0.000u 0:00.000`
